roxie/ccd/ccdserver.cpp

```diff
@@ -26987,9 +26987,50 @@ class CRoxieServerSoapActivityBase : public CRoxieServerActivity, implements ISo
     virtual void checkForAbort() { checkAbort(); }
 };
 
+class CRoxieServerSoapActionBase : public CRoxieServerSoapActivityBase
+{
+public:
+    CRoxieServerSoapActionBase(IRoxieAgentContext *_ctx, const IRoxieServerActivityFactory *_factory, IProbeManager *_probeManager)
+        : CRoxieServerSoapActivityBase(_ctx, _factory, _probeManager)
+    {
+    }
+
+    virtual void execute(unsigned parentExtractSize, const byte * parentExtract)
+    {
+        CriticalBlock b(ecrit);
+        if (exception)
+            throw(exception.getLink());
+        if (!executed)
+        {
+            try
+            {
+                onExecute(parentExtractSize, parentExtract);
```

Comment:
this isn't call start/stop() unlike CRoxieServerInternalSinkActivity.  Is that deliberate.  I think that may mean dependencies will not be executed.

Reply:
It's all a bit of a mess - the start/stop are in one (but not both) of the onExecutes being called...

I suspect that means that one of them had a bug previously...